### PR TITLE
upgrade all deps

### PR DESCRIPTION
**Pull request by yoshuawuyts**
Upgrades all dependencies. Thanks!

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -19,16 +19,16 @@ Event loop and I/O resources for asynchronous network services using futures
...
-parking_lot = "0.6.3"
-slab = "0.4.0"
-libc = "0.2.43"
+num_cpus = "1.10.0"
+parking_lot = "0.7.1"
+slab = "0.4.2"
+libc = "0.2.48"
 
 [dependencies.futures]
 version = "0.3.0-alpha.11"
```
Comment (taiki-e) [inline Cargo.toml]: I forgot to update this in #66...

Comment (kpp) [reply Cargo.toml]: Yes, how about pinning to exact `=0.3.0-alpha.13`?

Comment (taiki-e) [reply Cargo.toml]: Sounds good, futures-rs is also using it (https://github.com/rust-lang-nursery/futures-rs/pull/1412).

**Comment by PvdBerg1998** [comment]
Recently `LocalWaker` got renamed to `Waker`, this should get updated also

**Comment by kpp** [comment]
@PvdBerg1998 already done in https://github.com/withoutboats/romio/pull/66

**Comment by PvdBerg1998** [comment]
I'm getting errors that `tcp::TcpStream` does not implement `AsyncRead/Write`, this might be because Romio is still on futures alpha 11?

**Pull request merged**